### DISCUSSION TITLE
(Question) High Availability with Replication in EKS setup?

### DISCUSSION THREAD

**Discussion by marc-philipp-knechtle** [arkmq-org/arkmq-org-broker-operator#998] [General] [answered]
Hi, 

I'm using this project to deploy Artemis on EKS. 
I've wondered if it's possible to configure [HA-Replication](https://activemq.apache.org/components/artemis/documentation/latest/ha#replication) through the operator. I've not found any comments on this besides the [Openshift Overview of high availability](https://docs.redhat.com/en/documentation/red_hat_amq_broker/7.11/html-single/deploying_amq_broker_on_openshift/index#con-br-high-availability_broker-ocp)
This approach depends on the Persistent Volumens to achieve High Availability. However, I'd like to use the Broker's internal replication mechanism. Is there anybody who tried this already? Can this be configured through the `brokerProperties`? 

Thanks a lot in advance!

**Comment by brusdev** [comment, marked as answer]
Hi @marc-philipp-knechtle, the ArtemisCloud Opertor supports leader/follower JDBC HA, see https://github.com/artemiscloud/activemq-artemis-operator/blob/main/controllers/activemqartemis_jdbc_ha_test.go

Maybe you could also deploy a similar solution by using the right storage class for a shared persistent volume but I have never tried.

AFAIK there are no plans to improve the support for replication mechanism. 

Could you clarify why the approach that depends on the Persistent Volumes to provide HA doesn't work for your use case?

**Reply by marc-philipp-knechtle** [reply]
Hi @brusdev , 

thanks a lot for your answer :) 

I wanted to deploy the operator with PV from the instances [EKS Persistent Volumes for Instance Store](https://aws.amazon.com/blogs/containers/eks-persistent-volumes-for-instance-store/). But noticed that this breaks the High Availability as the Persistent Volumes fail once a EC2 instance fails. Because of this, I tried to configure the replication mechanism. 

A shared storage solution would again rely on an external mechanism to provide the high availability. When using Instance Store, a node failure seems pretty bad. So this is sadly not a solution. 

I think I'll stay on the EBS version for HA. 

Thank You!

**Reply by jwgmeligmeyling** [reply]
@brusdev Interesting, I was looking for this. Is the ArtemisCloud JDBC HA documented somewhere other than the test case?

Are there benefits for JDBC HA over shared storage HA? Why is JDBC HA supported / included in the test suite but PV HA not?

**Reply by brusdev** [reply]
@jwgmeligmeyling there is no doc about the ArtemisCloud JDBC HA solution. The main benefits of this solution are:
- short outage window when a node crash
- HA on multiple availability zones with a DB that supports HA on multiple availability zones

There is also a HA test by using a PV, see https://github.com/artemiscloud/activemq-artemis-operator/blob/main/controllers/activemqartemis_rwm_pvc_ha_test.go
